File: backend/src/main/java/at/ac/tuwien/sepr/assignment/individual/entity/Horse.java
package at.ac.tuwien.sepr.assignment.individual.entity;

import at.ac.tuwien.sepr.assignment.individual.type.Sex;
import java.time.LocalDate;

/**
 * Represents a horse in the persistent data store.
 */
public class Horse {
  private Long id;
  private String name;
  private Sex sex;
  private LocalDate dateOfBirth;
  private float height;
  private float weight;
  private Long breedId;

  public Long getId() {
    return id;
  }

  public Horse setId(Long id) {
    this.id = id;
    return this;
  }

  public String getName() {
    return name;
  }

  public Horse setName(String name) {
    this.name = name;
    return this;
  }

  public Sex getSex() {
    return sex;
  }

  public Horse setSex(Sex sex) {
    this.sex = sex;
    return this;
  }

  public LocalDate getDateOfBirth() {
    return dateOfBirth;
  }

  public Horse setDateOfBirth(LocalDate dateOfBirth) {
    this.dateOfBirth = dateOfBirth;
    return this;
  }

  public float getHeight() {
    return height;
  }

  public Horse setHeight(float height) {
    this.height = height;
    return this;
  }

  public float getWeight() {
    return weight;
  }

  public Horse setWeight(float weight) {
    this.weight = weight;
    return this;
  }

  public Long getBreedId() {
    return breedId;
  }

  public Horse setBreedId(Long breedId) {
    this.breedId = breedId;
    return this;
  }

  @Override
  public String toString() {
    return "Horse{"
        + "id=" + id
        + ", name='" + name + '\''
        + ", sex=" + sex
        + ", dateOfBirth=" + dateOfBirth
        + ", height=" + height
        + ", weight=" + weight
        + ", breed=" + breedId
        + '}';
  }
}

File: frontend/src/app/dto/horse.ts
import {Sex} from './sex';
import {Breed} from "./breed";

export interface Horse {
  id?: number;
  name: string;
  sex: Sex;
  dateOfBirth: Date;
  height: number;
  weight: number;
  breed?: Breed;
}

export interface HorseListDto {
  id: number,
  name: string,
  sex: Sex,
  dateOfBirth: Date;
  breed: Breed;
}


export interface HorseSearch {
  name?: string;
  sex?: Sex;
  bornEarliest?: Date;
  bornLastest?: Date;
  breedName?: string;
  limit?: number;
}

export interface HorseSelection {
    id: number;
    name: string;
    dateOfBirth: Date;
}
